### ISSUE TITLE
False positive: await-async-query

### ISSUE THREAD

**Issue by josias-r** [testing-library/eslint-plugin-testing-library#534]
### Plugin version

v5.0.3

### ESLint version

v8.6.0

### Node.js version

v16.13.1

### npm/yarn version

v8.1.2

### Operating system

macOS Monterey 12.1

### Bug description

The `await-async-query` fires a false positive for `findByProps` (used with react).

More details: https://github.com/facebook/react/issues/23093

### Steps to reproduce

1. Create react app with react-scripts v5
2. Create a test which makes use of `findByProps`
3. Run lint -> causes false positive error

### Error output/screenshots

https://github.com/facebook/react/issues/23093

### ESLint configuration

```
{
  "extends": ["react-app", "react-app/jest", "plugin:prettier/recommended"]
}
```

### Rule(s) affected

`await-async-query`

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
Thanks for reporting @josias-r!

I'm afraid this is related to Aggressive Reporting. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior?

**Comment by josias-r** [comment]
@Belco90 So am I understanding it correctly, that the `eslint-plugin-testing-library` has nothing to do with the react implementation?
So the react testing library is not based on this one?

If they don't relate, I can look into fixing it with one of the proposed fixes from troubleshoting. Otherwise, if the react testing-library is based on your eslint-plugin, it would make sense to fix it on their side, wouldn't it? 🤔

**Comment by Belco90** [comment]
@josias-r I think I'm not following you. `react-testing-library` was created first, then the core `dom-testing-library` was extracted, and finally packages for other frameworks were created too (e.g. `angular-testing-library`).

After this, `eslint-plugin-testing-library` was created to cover some of the functionality for a few Testing Library packages, and has been updated reflecting changes and new features from Testing Library.

They do relate of course. The problem you have with the plugin is that it tries to be too smart to identify Testing Library queries besides the built-in ones. You can restrict that behavior with the approaches mentioned in the Troubleshooting section.

**Comment by josias-r** [comment]
ah okay, I was not aware of how these packages relate to each other. I was just using `create-react-app` and confused about having a linting error about a react-related method whilst using the official way of bootstrapping a react app which they recommend.

The troubleshoot section seems to exactly address the problem I had, thank you

**Comment by Belco90** [comment]
I'm glad to hear that! Don't hesitate to open more issues/discussions if have more questions.
